vendor/assets/stylesheets/pure_css_reset.css

```diff
@@ -90,6 +90,18 @@ h6 {
   top: -1px;
 }
 
+.pure-button-group {
+  white-space: nowrap;
+}
+
+.pure-button-group .pure-button {
+  margin-right: -0.3em;
```

Comment:
This sounds like it could be a spacing specific to the admin theme. Investigate the `word-spacing` and similar properties for collapsing the space between buttons. You may have to apply that to `.pure-button-group` and then reset it on `.pure-button-group .pure-button`.